lib/Dialect/TTIR/IR/TTIROps.cpp

```diff
@@ -2,258 +2,71 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-#include <cstdint>
-#include <string>
+#include "ttmlir/Dialect/TTIR/IR/TTIROps.h"
```

Comment:
It stills seems a little random. For example, I would've expected to see conv2d and maxpool ops near each other. Can we group the ops by their functionality?

E.g.:
- memory/layout ops (tolayout, tomemcfg, todevice, fromdevice, ...)
- tensor creation ops (empty, full, ones, zeros, ...)
- eltwise unary ops
- eltwise binary ops
- matmul ops
- reduction ops
- conv ops (conv2d, maxpool, avgpool, ...)
- TM ops (transpose, concat, reshape, slice, ...)

Reply:
> It stills seems a little random. For example, I would've expected to see conv2d and maxpool ops near each other. Can we group the ops by their functionality?
> 
> E.g.:
> 
> * memory/layout ops (tolayout, tomemcfg, todevice, fromdevice, ...)
> * tensor creation ops (empty, full, ones, zeros, ...)
> * eltwise unary ops
> * eltwise binary ops
> * matmul ops
> * reduction ops
> * conv ops (conv2d, maxpool, avgpool, ...)
> * TM ops (transpose, concat, reshape, slice, ...)

That was the idea :) Currently, ops are divided in TTIR:
- Conv
- DataMovement(TMs)
- Layout
- Matmul
- MemoryManagement(alloc)
- Normalization(Softmax)
- Pooling
- Generic
- Kernel

I will merge pooling under the Conv category, that makes sense. Groups are in alphabetic order except Generic\Kernel, they seem to be related, hence I put them at the end. Maybe it makes sense to extract Generic and Kernel ops implementation into a separate file...

Reply:
Ah, I guess I first went to check for conv&maxpool, thanks for letting me know!